### PR TITLE
Fix Cache Tree Cache Misses

### PR DESCRIPTION
**Pull request by kyleve**
Due to float addition and subtraction that happens throughout layout, it was common to end up with `210.0` getting turned into `209.999997` (etc) throughout. This was causing needless cache misses. To resolve, I introduced pixel-based rounding to the cache tree so the cached constraint keys land on a display pixel.

### CODE REVIEW

File: BlueprintUI/Sources/Internal/CacheTree.swift
Comment (kyleve) [inline BlueprintUI/Sources/Internal/CacheTree.swift:37]: I know why this breaks; just rounding the key we use is correct; we don't want to round the constraint itself. Remove this TODO and rename `roundedConstraint` to `key`

**Review by n8chur — APPROVED** [review]
Looks reasonable to me, but would appreciate a review from @watt before this lands!

File: BlueprintUI/Tests/RenderPassCacheTests.swift
Comment (n8chur) [inline BlueprintUI/Tests/RenderPassCacheTests.swift:5]: Should we add a test that validates the rounding logic is cached appropriately?

Comment (kyleve) [reply BlueprintUI/Tests/RenderPassCacheTests.swift:5]: Probably a good idea! Will do.

**Review by nononoah — APPROVED** [review]
Seems like a nice easy win here, unless of course rounding fails to pay for itself in cache hits.

I'll defer to Andrew as the expert here, but it might be nice to show as part of PRs like this one that performance has improved as a result.

File: BlueprintUI/Sources/Internal/CacheTree.swift
Comment (nononoah) [inline BlueprintUI/Sources/Internal/CacheTree.swift:38]: Since we don't actually update the constraint, how important is the `screenScale` rounding?

**Pull request closed without merging**
